packages/viewer/src/components/viewer-pin-label/vertex-pin-label.tsx

```diff
@@ -0,0 +1,242 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  h,
+  Host,
+  Method,
+  Prop,
+  State,
+  Watch,
+} from '@stencil/core';
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+import { Dimensions, Point } from '@vertexvis/geometry';
+import { Disposable } from '@vertexvis/utils';
+import classNames from 'classnames';
+
+import { getMouseClientPosition } from '../../lib/dom';
+import { PinController } from '../../lib/pins/controller';
+import { isTextPin, TextPin } from '../../lib/pins/entities';
+import {
+  translatePointToRelative,
+  translatePointToScreen,
+} from '../viewer-markup/utils';
+
+@Component({
+  tag: 'vertex-viewer-pin-label',
+  styleUrl: 'vertex-pin-label.css',
+  shadow: false,
+})
+export class VertexPinLabel {
+  /**
+   * The pin to draw for the group
+   */
+  @Prop({ mutable: true })
+  public pin?: TextPin;
+
+  /**
+   * The dimensions of the canvas for the pins
+   */
+  @Prop({ mutable: true })
+  public elementBounds?: DOMRect;
+
+  /**
+   * The dimensions of the the pin label
+   */
+  @Prop({ mutable: true })
+  public pinLabelDimensions: Dimensions.Dimensions = { height: 0, width: 0 };
+
+  /**
+   * The current text value of the component. Value is updated on user
+   * interaction.
+   */
+  @Prop({ mutable: true })
+  public value: string;
+
+  /**
+   * The controller that drives behavior for pin operations
+   */
+  @Prop()
+  public pinController?: PinController;
+
+  /**
+   * @internal
+   */
+  @Event({ bubbles: true })
+  public labelChanged!: EventEmitter<void>;
+
+  @Element()
+  private hostEl!: HTMLElement;
+
+  @State()
+  private focused = false;
+
+  private inputEl?: HTMLInputElement;
+
+  public constructor() {
+    if (this.pin?.attributes?.labelText != null) {
+      this.value = this.pin.attributes.labelText;
+    } else {
+      this.value = '';
+    }
+  }
+
+  /**
+   * Gives focus to the the component's internal text input.
+   */
+  @Method()
+  public async setFocus(): Promise<void> {
+    // HTMLInputElement.focus() doesn't exist in tests.
+    if (typeof this.inputEl?.focus === 'function') {
+      this.inputEl?.focus();
+    }
+  }
+
+  @Watch('focused')
+  protected watchFocusChange(): void {
+    this.labelChanged.emit();
+  }
+
+  protected componentDidLoad(): void {
+    const resize = new ResizeObserver(() => {
+      this.labelChanged.emit();
+    });
+
+    if (this.hostEl != null) {
+      resize.observe(this.hostEl);
+    }
```

Comment:
Could we add a `disconnectedCallback` that removes this resize observer as well for cleanup purposes?